corehq/apps/users/models.py

```diff
@@ -3022,4 +3023,6 @@ def role(self):
                 return UserRole.get(self.role_id)
             except ResourceNotFound:
                 logging.exception('no role with id %s found in domain %s' % (self.role_id, self.domain))
+        elif self.domain:
+            return CouchUser.from_django_user(self.user).get_domain_membership(self.domain).role
```

Comment:
When enterprise permissions are on, this will grant access to both their domain and any controlled domain (so, COVID state-level users will have access to all county domains). That's probably what you want, just wanted to confirm.

Reply:
I didn't consider this use case and don't know much about how the enterprise permissions work to have an opinion. Do you think it's the right behavior given the expectations of those roles as they're being used by the team?

Reply:
Yes, I think the current behavior matches what the team will expect.

Reply:
Having said that, I might as well verify that assumption. I'll report back if the current behavior is *not* what they expect.

Reply:
Can you report back either way? Would feel better about having the loop closed.

Reply:
The current code is the correct behavior.